src/pages/_middleware.ts

```diff
@@ -77,6 +91,19 @@ export function middleware(req: NextRequest, ev: NextFetchEvent) {
     return NextResponse.redirect(url, permanent ? 308 : 307)
   }
 
+  // Rewrite .io pages
```

Comment:
We might want to change the precedence in `determineProductSlug` with this change. I think the hostname check should take priority, before the cookie.

It's likely not a huge deal, as I would need to have an `io_preview` cookie set on packer.io that is NOT packer, for example, but seems like the correct ordering.